src/vm/analysis/type_checker/natives/sequences.rs

```diff
@@ -391,3 +391,39 @@ pub fn check_special_index_of(
 
     TypeSignature::new_option(TypeSignature::UIntType).map_err(|e| e.into())
 }
+
+pub fn check_special_slice(
+    checker: &mut TypeChecker,
+    args: &[SymbolicExpression],
+    context: &TypingContext,
+) -> TypeResult {
+    check_argument_count(3, args)?;
+
+    // Check sequence
+    let seq_type = checker.type_check(&args[0], context)?;
+    let seq = match &seq_type {
+        TypeSignature::SequenceType(seq) => TypeSignature::SequenceType(seq.clone()),
```

Comment:
Do we need to clone this?  Can we just use `seq_type` as-is?

Reply:
The error arm uses `seq_type`, which is why the match is done with `&seq_type`. So, that makes `seq` a `&SequenceSubtype`, which leads to the clone - is there a way around this?